data/src/main/java/io/gnosis/data/repositories/SafeRepository.kt

```diff
@@ -64,23 +64,56 @@ class SafeRepository(
     suspend fun clearActiveSafe() {
         preferencesManager.prefs.edit {
             remove(ACTIVE_SAFE)
+            remove(ACTIVE_SAFE_SIGNING_OWNERS)
         }
     }
 
     suspend fun setActiveSafe(safe: Safe) {
         preferencesManager.prefs.edit {
             putString(ACTIVE_SAFE, "${safe.address.asEthereumAddressString()};${safe.localName};${safe.chainId}")
+            putString(
+                ACTIVE_SAFE_SIGNING_OWNERS,
+                if (safe.signingOwners.isEmpty()) {
+                    null
+                } else {
+                    safe.signingOwners.joinToString(";") { it.asEthereumAddressString() }
+                }
+            )
         }
     }
 
     suspend fun getActiveSafe(): Safe? {
         val activeSafeData = preferencesManager.prefs.getString(ACTIVE_SAFE, null)?.split(";")
         val address = activeSafeData?.get(0)?.asEthereumAddress()
-        return address?.let {
+        val safe = address?.let {
             val chainId = nullOnThrow { activeSafeData[2].toBigInteger() }
             // safes from old version won't have chain id saved
             if (chainId != null) getSafeBy(it, chainId) else getSafeBy(it)
         }
+        safe?.signingOwners = getActiveSafeSigningOwners()
+        return safe
+    }
+
+    suspend fun setActiveSafeSigningOwners(owners: List<Solidity.Address>) {
+        preferencesManager.prefs.edit {
+            putString(
+                ACTIVE_SAFE_SIGNING_OWNERS,
+                if(owners.isEmpty()) {
+                    null
+                } else {
+                    owners.joinToString(";") {
```

Comment:
Should ";" be extarcted as for example `ADDRESS_SEPARATOR` constant?

Reply:
yes, as it is used in multiple places